src/main/java/net/mcreator/blockly/java/blocks/ProcedureCallBlock.java

```diff
@@ -116,43 +96,82 @@ public class ProcedureCallBlock implements IBlockGenerator {
 			master.addCompileNote(new BlocklyCompileNote(BlocklyCompileNote.Type.WARNING,
 					L10N.t("blockly.warnings.call_procedure.empty")));
 		}
 	}
 
+	@Nonnull
+	static List<DependencyInput> mapDependencies(BlocklyToCode master, Element block, List<Dependency> dependencies,
+			List<String> skippedDepsNames, List<String> processedDepsNames) throws TemplateGeneratorException {
+		// The procedure dependencies, in a flattened {"name": "type"} map
+		Map<String, String> flattenedDeps = dependencies.stream()
+				.collect(Collectors.toMap(Dependency::getName, Dependency::getRawType));
+		List<DependencyInput> depInputs = new ArrayList<>();
+
+		Element mutation = XMLUtil.getFirstChildrenWithName(block, "mutation");
+		if (mutation != null && mutation.hasAttribute("inputs") && !mutation.getAttribute("inputs")
+				.equals("undefined")) {
+			int depCount = Integer.parseInt(mutation.getAttribute("inputs"));
+			Map<String, Element> fields = XMLUtil.getChildrenWithName(block, "field").stream()
+					.filter(e -> e.getAttribute("name").matches("name\\d+"))
+					.collect(Collectors.toMap(e -> e.getAttribute("name"), e -> e));
+			Map<String, Element> inputs = XMLUtil.getChildrenWithName(block, "value").stream()
+					.filter(e -> e.getAttribute("name").matches("arg\\d+"))
+					.collect(Collectors.toMap(e -> e.getAttribute("name"), e -> e));
+
+			for (int i = 0; i < depCount; i++) {
+				String currentName = fields.get("name" + i).getTextContent();
+				String currentArg;
+				if (inputs.containsKey("arg" + i)) {
+					// If the procedure actually has this dependency, also generate the code
+					if (flattenedDeps.containsKey(currentName)) {
+						currentArg = master.directProcessOutputBlockWithoutParentheses(inputs.get("arg" + i));
+					} else {
+						// We don't need to do any further processing, skip this dependency
+						skippedDepsNames.add(currentName);
+						continue;
+					}
+				} else {
+					// Keep processing to look for other missing inputs
+					currentArg = "";
+					master.addCompileNote(new BlocklyCompileNote(BlocklyCompileNote.Type.ERROR,
+							L10N.t("blockly.errors.call_procedure.missing_inputs", currentName)));
+				}
+				depInputs.add(new DependencyInput(currentName, flattenedDeps.get(currentName), currentArg));
+				processedDepsNames.add(currentName);
+			}
+		}
+
+		// Add to master all the dependencies that weren't processed
+		dependencies.stream().filter(e -> !processedDepsNames.contains(e.getName())).forEach(master::addDependency);
+
+		return depInputs;
+	}
+
 	@Override public String[] getSupportedBlocks() {
-		return new String[] { "call_procedure", "call_procedure_at", "old_command" };
+		return new String[] { "call_procedure", "old_command" };
 	}
 
 	@Override public BlockType getBlockType() {
 		return BlockType.PROCEDURAL;
 	}
+
+	// The record holds info about a single dependency row in the block (name, type, input code)
+	public record DependencyInput(String name, String type, String arg) {}

```

Comment:
`arg` should be called `value`. 

The whole DependencyInput object is "(method) argument", while what you currently call "arg" is actually "argument value"

Reply:
If what, this is not my code.